sw/device/tests/dif/dif_otbn_sanitytest.c

```diff
@@ -0,0 +1,252 @@
+
+  CHECK(app->dmem_end >= app->dmem_start);
+  const size_t dmem_size = app->dmem_end - app->dmem_start;
+
+  LOG_INFO(
+      "Loading OTBN instruction memory image stored between address 0x%x and "
+      "%p (%d bytes)",
+      app->imem_start, app->imem_end, imem_size);
+  CHECK(imem_size > 0);
+  CHECK(imem_size % 4 == 0);
+  CHECK(dif_otbn_imem_write(&otbn, 0, (uint32_t *)app->imem_start, imem_size) ==
+            kDifOtbnOk,
+        "Unable to write IMEM application image (.text) to OTBN.");
+
+  if (dmem_size > 0) {
+    LOG_INFO(
+        "Loading OTBN data memory image stored between address %p and %p (%d "
+        "bytes)",
+        app->dmem_start, app->dmem_end, dmem_size);
+    CHECK(dif_otbn_dmem_write(&otbn, 0, (uint32_t *)app->dmem_start,
+                              dmem_size) == kDifOtbnOk,
+          "Unable to write DMEM application image (.data) to OTBN.");
+  } else {
+    LOG_INFO("No OTBN data memory image to load.");
+  }
+}
+
+/**
+ * Call a function on OTBN
+ *
+ * Set the entry point (start address) of OTBN to the desired function, and
+ * starts the OTBN operation.
+ *
+ * @param func the function to be called
+ */
+static void call_function(const otbn_func_t *func) {
+  uint32_t start_address = func->func - func->app->imem_start;
+  LOG_INFO("Calling function at address 0x%x on OTBN.", start_address);
+  CHECK(dif_otbn_start(&otbn, start_address) == kDifOtbnOk);
+}
+
+/**
+ * Busy wait for OTBN to be done with its operation.
+ */
+static void otbn_wait_for_done() {
+  bool busy = true;
+
+  while (busy) {
+    CHECK(dif_otbn_is_busy(&otbn, &busy) == kDifOtbnOk,
+          "Unable to get busy status from OTBN");
+  }
+}
+
+/**
+ * Initialize OTBN's data memory with zeros
+ */
+static void zero_dmem() {
+  const uint32_t zero = 0;
+  int dmem_size_words = dif_otbn_get_dmem_size_bytes(&otbn) / 4;
+  for (int i = 0; i < dmem_size_words; ++i) {
+    dif_otbn_result_t rv =
+        dif_otbn_dmem_write(&otbn, i * 4, &zero, sizeof(zero));
+    CHECK(rv == kDifOtbnOk, "Error zeroing word %d in OTBN DMEM: %d", i, rv);
+  }
+}
+
+/**
+ * Run a 384-bit Barrett Multiplication on OTBN and check its result.
+ *
+ * This test is not aiming to exhaustively test the Barrett multiplication
+ * itself, but test the interaction between device software and OTBN. As such,
+ * only trivial parameters are used.
+ *
+ * The code executed on OTBN can be found in sw/otbn/code-snippets/barrett384.s.
+ * The entry point wrap_barrett384() is called according to the calling
+ * convention described in the OTBN assembly code file.
+ */
+static void test_barrett384() {
+  zero_dmem();
+
+  load_app(&kOtbnAppBarrett384);
+
+  // a, first operand
+  const uint8_t a[48] = {10};
+
+  // b, second operand
+  const uint8_t b[48] = {20};
+
+  // m, modulus, max. length 384 bit with 2^384 > m > 2^383
+  // We choose the modulus of P-384: m = 2**384 - 2**128 - 2**96 + 2**32 - 1
+  const uint8_t m[48] = {
```

Comment:
Single letter variables should be avoided when they have a better name available, like `kModulus` and `kBarrettConstant`.

(Also, since these are morally constants, they aught to have kConst names. I would make them static.)

Reply:
These are locals, and they do correspond to the names in the assembly routine.